Cellular: Documenting the support for Non-IP data delivery feature

mirelachirica opened the pull request:
@lauri-piikivi, @AriParkkila please review.

Commit:
Cellular: Documenting the support for Non-IP data delivery feature

AriParkkila reviewed (approved):
Need to update cellular configuration docs as well?

AnotherButler commented:
Thanks for the PR, @mirelachirica 
It looks like a code snippet is failing. Could you please look into this?

mirelachirica commented:
> It looks like a code snippet is failing. Could you please look into this?

I have commited one change I think will fix the problem. Where to find the relevant scripts to verify these changes locally? thanks!

mirelachirica commented:
> Need to update cellular configuration docs as well?

I updated it.

AnotherButler commented:
@mirelachirica Thanks 👍 It's all passing now. 

You can find the files here: https://github.com/ARMmbed/mbed-os-5-docs/blob/development/.travis.yml and https://github.com/ARMmbed/mbed-os-5-docs/blob/development/check_tools/find_bad_code_snippets.sh

AnotherButler commented:
@lauri-piikivi Is this ready to merge?

lauri-piikivi commented on docs/api/networksocket/CellularNonIPSocket.md:
```diff
@@ -0,0 +1,83 @@
+## Non-IP cellular socket
+
+The CellularNonIPSocket class provides, through standard socket send and recv member functions, the ability to send and receive Non-IP datagrams using Control Plane CIoT EPS optimisation feature of the NB-IoT networks. This feature is implemented in ControlPlane_netif class. [`ControlPlane_netif`](https://os.mbed.com/docs/development/mbed-os-api-doxy/classmbed_1_1_control_plane__netif.html) class ...
```
State here "send and receive the 3GPP Non-IP datagrams (NIDD)"

lauri-piikivi replied:
Remove the Control Plane CIoT EPS --- scary :) This could be removed in many places, just say "Cellular IoT" ?

mirelachirica replied:
done

lauri-piikivi commented on docs/api/networksocket/CellularNonIPSocket.md:
```diff
@@ -0,0 +1,83 @@
+The CellularNonIPSocket class provides, through standard socket send and recv member functions, the ability to send and receive Non-IP datagrams using Control Plane CIoT EPS optimisation feature of the NB-IoT networks. This feature is implemented in ControlPlane_netif class. [`ControlPlane_netif`](https://os.mbed.com/docs/development/mbed-os-api-doxy/classmbed_1_1_control_plane__netif.html) class ...
+
+The constructor takes no parameters. To initialize the socket on a specified NetworkInterface, you must call `open` method, which takes a CellularContext pointer.
+
```
Is this inline with the multi-homing socket operation, where they give the interface name? Could check with them

mirelachirica replied:
I will update this when/if CellularNonIPSocket needs to be changed due to multi-homing feature. This reflects current implementation.

lauri-piikivi commented on docs/api/networksocket/CellularNonIPSocket.md:
```diff
@@ -0,0 +1,83 @@
+The constructor takes no parameters. To initialize the socket on a specified NetworkInterface, you must call `open` method, which takes a CellularContext pointer.
+
+[`CellularContext`](https://os.mbed.com/docs/development/mbed-os-api-doxy/_cellular_context_8h.html) is setting up the modem into the Control Plane CIoT EPS optimisation mode of operation if it is requested and cellular network supports it.
+
```
add
"Control plane optimisations is a new feature for Cellular IoT (CIoT), where the device can use cellular control channels for data communications, to save power and resources as there is less radio singling needed. 

NOTE! Non-IP usage usually requires integration to cellular operator messaging service. The service supports a web API to send and receive to the non-IP using devices.  "

mirelachirica replied:
done

lauri-piikivi commented on docs/reference/technology/connectivity/connectivity.md:
```diff
@@ -101,6 +102,12 @@ Near-field communication (NFC) is a short range (few centimeters) wireless techn
+#### NB-IoT Cellular
+
+Non-IP Data Delivery(NIDD) is a new feature for communication over NB-IoT. It is enabled by Control Plane CIoT EPS optimization and meant to provide improved support of small data transfer. It does this by transporting user data encapsulated in NAS messages, thus reducing the total number of control plane messages when handling a short data transaction.
+
```
"user data over the control channel", no need to mention the NAS here.

mirelachirica replied:
done

lauri-piikivi reviewed (changes requested):
I just wanted to remove some of the scary words and acronyms, otherwise I think it was very good!

mirelachirica commented:
@lauri-piikivi can you review

mirelachirica commented:
Closed by accident

mirelachirica commented:
@lauri-piikivi is this ok as it is?

mirelachirica commented:
The NIDD suport is now in master, introduced by feature-cellular-refactor merge to master:
https://github.com/ARMmbed/mbed-os/commit/8c2ad14acecac4ea5151f0fc7e94ca4263efa287

AnotherButler commented:
@lauri-piikivi Have your concerns been addressed?

Commit:
Edit CellularNonIPSocket.md

Edit file, mostly for minor style change and international English (as opposed to UK English).

AnotherButler commented on docs/api/networksocket/CellularNonIPSocket.md:
```diff
@@ -0,0 +1,88 @@
+
+[`CellularContext`](https://os.mbed.com/docs/development/mbed-os-api-doxy/_cellular_context_8h.html) sets up the modem into the Control Plane optimization mode of operation if it is requested and if the cellular network supports it.
+
+Control plane optimization is a new feature for cellular IoT (CIoT). With it, the device can use cellular control channels for data communications, to save power and resources. It does this by using less radio singling.
```
Is "radio singling" correct, or is it "radio signaling"?

mirelachirica replied:
signaling is the correct term

mirelachirica replied:
About the rephrasing. Is it clear that "this" in "It does **this** by using" refers to saving power and resources?

AnotherButler replied:
No, it's not clear. Good catch. What do you think about "...to save power and resources by using less radio signaling."?

mirelachirica replied:
Sounds good!

AnotherButler reviewed (commented):
Please answer the question I left and review my changes to make sure I didn't accidentally change the meaning of anything.

Commit:
Edit networksocket.md

Edit file for consistent capitalization across docs.

Commit:
Edit connectivity.md

Make minor edits to add a missing space and standardize capitalization in headings throughout the document.

mirelachirica commented on docs/api/networksocket/networksocket.md:
```diff
@@ -97,7 +97,7 @@ The network socket API provides a common interface for using sockets on network
 - [TCPSocket](tcpsocket.html): This class provides the ability to send a stream of data over TCP.
 - [SocketAddress](socketaddress.html): You can use this class to represent the IP address and port pair of a unique network endpoint.
-- [CellularNonIPSocket](cellularnonipsocket.html): This class provides the ability to send and receive 3GPP Non-IP datagrams (NIDD) using Cellular IoT feature.
+- [CellularNonIPSocket](cellularnonipsocket.html): This class provides the ability to send and receive 3GPP non-IP datagrams (NIDD) using our cellular IoT feature.
```
Cellular -> cellular is fine but "our" I dont think so. It was meant as IoT feature for cellular networks, so a cellluar IoT feature not mbed cellular IoT feature.

AnotherButler replied:
Thanks for explaining. Would "...using the cellular IoT feature." be better?

mirelachirica replied:
yes

Commit:
Update CellularNonIPSocket.md

Make change suggested in comments.

Commit:
Update networksocket.md

Make change suggested in comments for accuracy.

AnotherButler commented:
To clarify, this is only for 5.12, right? Not for 5.11?

mirelachirica commented:
> To clarify, this is only for 5.12, right? Not for 5.11?
Yes, for 5.12 only.

AnotherButler commented:
@lauri-piikivi I'm going to merge this. If you have any further changes, please raise them in a new PR.
